Introducing mvpage, a command used to move markdown pages across directories for istio.io

geeknoid opened the pull request.

mandarjog commented:
Did you mean to include the binary?
Apart from that looks ok.
Also would this be a PR in Hugo itself instead of here?

mandarjog commented on cmd/mvpage/main.go:
```diff
@@ -0,0 +1,63 @@
+	rootCmd := cobra.Command{
+		Use:   "mvpage <old markdown file> <new markdown file>",
+		Short: "Moves a content page within a Hugo site.",
+		Long: "Moves a content page within a Hugo site, updating all links to the page\n" +
```
Can you add the usage content to a readme.md in this directory?

geeknoid replied:
Done.

mandarjog commented:
The only other observation is that just doing “mv” sometimes loses its git history.
A git mv, update contents , git add ensures that git history is preserved.
Consider either adding that or making it possible .

istio-testing commented:
@geeknoid: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
tools-lint | 827167609232d28da629dc358227bc516dd81e73 | [link](https://prow.istio.io/view/gcs/istio-prow/pr-logs/pull/istio_tools/263/tools-lint/17) | `/test tools-lint`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>
<!-- test report -->

geeknoid commented:
I deleted the binary and added a README file.

I'll contact the Hugo author to see if he'd be interested in the tool. It's generic enough that it should work for any  Hugo installation.

I'll look at adding 'git mv' functionality, that's a good idea.

PTAL.

duderino commented on cmd/mvpage/mover.go:
```diff
@@ -0,0 +1,188 @@
+		return err
+	}
+
+	if m.srcLinkRegex, err = regexp.Compile(regexp.QuoteMeta("(" + m.srcLink)); err != nil {
```
should you match the closing ')' too?  otherwise you might replace a prefix of a different link?

geeknoid replied:
I wanted to avoid problems with trailing fragments.  That  is (link#fragment). I guess I could incorporate optional fragments into the regex.
